d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -659,13 +727,18 @@ private void onData(ResourceUpdate data, XdsServerMetricsProvider metricsProvide
           _log.warn("Received invalid data for {} {}, data: {}", _type, _resource, data);
         }
         _xdsClientJmx.incrementResourceInvalidCount();
-      }
 
-      if (_data == null)
-      {
-        _log.info("Initializing {} {} to empty data.", _type, _resource);
-        _data = _type.emptyData();
+        if (_data == null)
+        {
+          _log.info("Initializing {} {} to empty data.", _type, _resource);
+          _data = _type.emptyData();
+        }
+        else {
+          // no update to the existing data, don't need to notify the watcher
+          return;
+        }
```

Comment:
```suggestion
        else
        {
          // no update to the existing data, don't need to notify the watcher
          return;
        }
```

Reply:
updated all "{" in this file that were on a new line.